lib/CryptoLD.js

```diff
@@ -67,10 +67,56 @@ class CryptoLD {
     return this.suites.get(type).from(serialized);
   }
 
+  /**
+   * Imports a key pair instance via the provided `documentLoader` function,
+   * optionally checking it for revocation and required context.
+   *
+   * @param {object} options - Options hashmap.
+   * @param {string} options.id - Key ID or URI.
+   * @param {Function} options.documentLoader - JSON-LD Document Loader.
+   * @param {boolean} [options.checkContext=true] - Whether to check that the
+   *   fetched key document contains the context required by the key's crypto
+   *   suite.
+   * @param {boolean} [options.checkRevoked=true] - Whether to check the key
+   *   object for the presence of the `revoked` timestamp.
+   *
+   * @returns {Promise<LDKeyPair>} Resolves with the appropriate key pair
+   *   instance.
+   */
+  async fromKeyId({
+    id, documentLoader, checkContext = true, checkRevoked = true
+  } = {}) {
+    if(!id) {
+      throw new TypeError('The "id" parameter is required.');
+    }
+    if(!documentLoader) {
+      throw new TypeError('The "documentLoader" parameter is required.');
+    }
+    let keyDocument;
+    try {
+      ({doc: keyDocument} = await documentLoader(id));
```

Comment:
The return signature of a document loader returns `document` not `doc`.

https://github.com/digitalbazaar/jsonld-document-loader/blob/master/lib/JsonLdDocumentLoader.js#L24-L29